lib/index.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Bedrock Agreement HTTP Module.
+ *
+ * Copyright (c) 2016 Digital Bazaar, Inc. All rights reserved.
+ */
+/* jshint node: true */
+var bedrock = require('bedrock');
+var brAgreement = require('bedrock-agreement');
+var brRest = require('bedrock-rest');
+var brPassport = require('bedrock-passport');
+var validate = require('bedrock-validation').validate;
+
+var BedrockError = bedrock.util.BedrockError;
+var ensureAuthenticated = brPassport.ensureAuthenticated;
+
+// load config defaults
+require('./config');
+
+bedrock.events.on('bedrock.test.configure', function() {
+  require('./test.config');
+});
+
+// add routes
+bedrock.events.on('bedrock-express.configure.routes', addRoutes);
+
+function addRoutes(app) {
+  var basePath = bedrock.config['bedrock-agreement-http'].basePath;
+
+  // get agreements for the authenticated identity
+  app.get(
+    basePath, ensureAuthenticated, brRest.when.prefers.ld,
+    brRest.makeResourceHandler({
+      get: function(req, res, callback) {
+        var actor = req.user ? req.user.identity : undefined;
+        brAgreement.get(actor, callback);
+      }
+    }));
+
+  // create a new agreement for the authenticated identity
+  app.post(
+    basePath, ensureAuthenticated, validate('services.agreement.postAgreement'),
+    function(req, res, next) {
+    var actor = req.user ? req.user.identity : undefined;
+    brAgreement.add(actor, req.body.agreement, function(err, result) {
+      if(err) {
+        return next(new BedrockError(
+          'The agreement could not be added.',
+          'AddAgreementFailed', {
+            httpStatusCode: 400,
+            'public': true
+          }, err));
+      }
+      res.status(201).json(result.event);
+    });
+  });
+}
+
+// add agrements to the session data
+bedrock.events.on(
+  'bedrock-session-http.session.get', function(req, session, callback) {
+  if(req.isAuthenticated() && req.user.identity) {
+    session.identity = session.identity || {};
+    return brAgreement.get(req.user.identity, function(err, agreements) {
+      if(err) {
+        return callback(err);
+      }
+      session.identity.agreements = agreements;
```

Comment:
I wonder if we should put this under `identity` or not. Maybe it should be top-level? `session.acceptedAgreements`?